pkg/reconciler/v1alpha1/autoscaling/kpa_scaler.go

```diff
@@ -84,6 +86,46 @@ func (ks *kpaScaler) getAutoscalerConfig() *autoscaler.Config {
 	return ks.autoscalerConfig.DeepCopy()
 }
 
+func scaleBoundInt32(ctx context.Context, kpa *kpa.PodAutoscaler, key string) int32 {
+	logger := logging.FromContext(ctx)
+	if s, ok := kpa.Annotations[key]; ok {
+		i, err := strconv.ParseInt(s, 10, 32)
+		if err != nil || i < 0 {
+			logger.Debugf("value of %s must be positive integer", key)
+			return 0
+		}
+		return int32(i)
+	}
+	return 0
+}
+
+func getScaleBounds(ctx context.Context, kpa *kpa.PodAutoscaler) (int32, int32) {
+	logger := logging.FromContext(ctx)
+
+	min := scaleBoundInt32(ctx, kpa, autoscaling.MinScaleAnnotationKey)
+	max := scaleBoundInt32(ctx, kpa, autoscaling.MaxScaleAnnotationKey)
+
+	if max != 0 && min > max {
+		logger.Debugf("%s must be less than or equal %s, adjusting min = max", autoscaling.MinScaleAnnotationKey, autoscaling.MaxScaleAnnotationKey)
+		min = max
+	}
+
+	return min, max
+}
+
+// pre: 0 <= min <= max && 0 <= x
+func applyBounds(min, max int32) func(int32) int32 {
+	return func(x int32) int32 {
+		if min > 0 && x < min {
```

Comment:
Since `scaleBoundInt32` always returns a non-negative value, why couldn't this test be replaced by:
```
if x < min {
```
?

Reply:
You're right! Fixin'

Reply:
Thanks.